### PR TITLE
Various improvements for usage with jupyter notebooks

### PR DESCRIPTION
**Pull request by pluegh**

### CODE REVIEW

**Comment by pluegh** [comment]
The tests fail:
```
Traceback (most recent call last):
  File "/usr/bin/eos-analysis", line 487, in <module>
    main()
  File "/usr/bin/eos-analysis", line 259, in main
    args.cmd(args)
  File "/usr/bin/eos-analysis", line 391, in cmd_sample_mcmc
    samples, weights = analysis.sample(N=args.N, stride=args.stride, pre_N=args.pre_N, preruns=args.preruns, rng=rng)
  File "/usr/lib/python3.6/dist-packages/eos/analysis.py", line 283, in sample
    for i in progressbar(range(0, preruns), desc="Pre-runs", leave=False):
TypeError: <lambda>() got an unexpected keyword argument 'desc'
```

Dammit, I did not test this outside of jupyter, where the progress bar skipped. Fix will be on the way.

File: python/eos/analysis.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ def sample_pmc(self, log_proposal, step_N=1000, steps=10, final_N=5000, rng=np.r
         eps = np.finfo(float).eps
 
         # carry out adaptions
-        for step in progressbar(range(steps)):
+        for step in progressbar(range(steps), desc="Adaptions", leave=False):
```
Comment (dvandyk) [inline python/eos/analysis.py:356]: Ditto, see above.

**Review by dvandyk — CHANGES_REQUESTED** [review]
Thank you very much @pluegh!

Changes requested:

1. For the weights, throw an appropriate error if both ``data['weights']`` and ``data['log_weights']`` are provided.
2. For the progress bar, please amend the commit message regarding vanishing progress bars.

File: python/eos/plot/plotter.py [hunk 2]
```diff
@@ -815,6 +815,7 @@ def __init__(self, plotter, item):
             if 'data' in item:
                 self.samples = item['data']['samples']
                 self.weights = np.exp(item['data']['log_weights']) if 'log_weights' in item['data'] else None
+                self.weights = item['data']['weights'] if 'weights' in item['data'] else None
```
[the remark this answers is not in the document]
Comment (dvandyk) [reply python/eos/plot/plotter.py:822]: This doesn't work either, does it? Now the weights will be set to ``None`` even if ``log_weights`` are provided.

File: python/eos/plot/plotter.py [hunk 3]
```diff
@@ -869,8 +874,13 @@ def __init__(self, plotter, item):
             self.samples = None
             self.weights = None
             if 'data' in item:
+
+                if 'weights' in item['data'] and 'log_weights' in item['data']:
+                    raise KeyError("Only one of 'weights' and 'log_weights' must be specified")
+
                 self.samples = item['data']['samples'].T
                 self.weights = np.exp(item['data']['log_weights']) if 'log_weights' in item['data'] else None
+                self.weights = item['data']['weights'] if 'weights' in item['data'] else None
```
Comment (dvandyk) [inline python/eos/plot/plotter.py:883]: Ditto, see above.

File: python/eos/plot/plotter.py [hunk 4]
```diff
@@ -971,8 +981,13 @@ def __init__(self, plotter, item):
             self.samples = None
             self.weights = None
             if 'data' in item:
+
+                if 'weights' in item['data'] and 'log_weights' in item['data']:
+                    raise KeyError("Only one of 'weights' and 'log_weights' must be specified")
+
                 self.samples = item['data']['samples']
                 self.weights = np.exp(item['data']['log_weights']) if 'log_weights' in item['data'] else None
+                self.weights = item['data']['weights'] if 'weights' in item['data'] else None
```
Comment (dvandyk) [inline python/eos/plot/plotter.py:990]: Ditto, see above.

**Review by dvandyk — CHANGES_REQUESTED** [review]
There is still a problem with the weights.

**Comment by pluegh** [comment]
Third time's the charm.

The code is very explicit (read: foolproof...) now and manually verified for all cases.

**Review by dvandyk — APPROVED** [review]
Thanks, that looks good!

**Pull request merged**